README.rst

```diff
@@ -10,3 +10,116 @@ flamingo
 
 Flamingo is a python3-based, `pelican <https://blog.getpelican.com/>`_-inspired
 static site generator.
+
+
+Basic Concepts
+--------------
+
+Flamingo aims to convert content written in a markup language into a static
+website. To archive this the flamingo core parses a directory of markup files
+and stores its content in an internal state called the ``context``. Plugins can
+manipulate the internal state before the ``context`` is used to generate static
+HTML using Jinja2 templates.
+
+Default directory structure
+---------------------------
+
+The following listing shows the default directory structure flamingo
+used if nothing else is configured:
+
+.. code-block:: none
+
+   |- settings.py
+   |- content
+   |- output
+   |  |- static
+   |- theme
+   |  |- static
+   |  |- templates
+
+settings.py
+...........
+
+This file is loaded by flamingo on startup.
+it can add, modify or remove the defaults provided by the core.
+
+
+content
+.......
+
+This directory contains the markup files with the actual content for the
+static site.
+Flamingo will scan all directories recursively.
+By default flamingo uses reStructuredText.
+
+Content files must follow a special format:
+
+.. code-block:: none
+
+   id: example
+   title: Title of this page
+   output: file_where_this_is_written.html
+   template: template_used.html
+   
+   
+   First headline
+   ==============
+   
+   The quick brown fox jumps over the lazy dog.
+
+The first section of the file contains metadata of the file.
+The next section contains the actual markup.
+Both sections must be separated by a double newline.
+
+For every file in the ``content`` directory a ``Content`` object in the
+``context.contents`` -set is created.
+All metadata are represented as items of the ``Content`` object.
+The markdown section is represented in the special item ``content_body``.
+
+theme
```

Comment:
All directories but `output/` are described, thus maybe add a short sectio for `output/`, too?